src/qt/pivx/masternodeswidget.cpp

```diff
@@ -197,29 +198,77 @@ void MasterNodesWidget::onEditMNClicked(){
     }
 }
 
-void MasterNodesWidget::startAlias(QString strAlias){
+void MasterNodesWidget::startAlias(QString strAlias) {
     QString strStatusHtml;
     strStatusHtml += "Alias: " + strAlias + " ";
 
     for (CMasternodeConfig::CMasternodeEntry mne : masternodeConfig.getEntries()) {
         if (mne.getAlias() == strAlias.toStdString()) {
             std::string strError;
-            CMasternodeBroadcast mnb;
-            if (CMasternodeBroadcast::Create(mne.getIp(), mne.getPrivKey(), mne.getTxHash(), mne.getOutputIndex(), strError, mnb)) {
-                strStatusHtml += "successfully started.";
-                mnodeman.UpdateMasternodeList(mnb);
-                mnb.Relay();
-                mnModel->updateMNList();
-            } else {
-                strStatusHtml += "failed to start.\nError: " + QString::fromStdString(strError);
-            }
+            strStatusHtml += (!startMN(mne, strError)) ? ("failed to start.\nError: " + QString::fromStdString(strError)) : "successfully started.";
             break;
         }
     }
-    inform(strStatusHtml);
+    // update UI and notify
+    updateModelAndInform(strStatusHtml);
+}
+
+void MasterNodesWidget::updateModelAndInform(QString informText) {
+    mnModel->updateMNList();
+    inform(informText);
+}
+
+bool MasterNodesWidget::startMN(CMasternodeConfig::CMasternodeEntry mne, std::string& strError) {
+    CMasternodeBroadcast mnb;
+    if (!CMasternodeBroadcast::Create(mne.getIp(), mne.getPrivKey(), mne.getTxHash(), mne.getOutputIndex(), strError, mnb))
+        return false;
+
+    mnodeman.UpdateMasternodeList(mnb);
+    mnb.Relay();
+    return true;
+}
+
+void MasterNodesWidget::onStartAllClicked() {
+    if(!verifyWalletUnlocked()) return;
+    if (isLoading) {
+        inform(tr("Start all Masternodes is being executed, please wait"));
+    } else {
+        isLoading = true;
+        if (!execute(REQUEST_START_ALL)) {
+            inform(tr("Cannot perform start all Mastenodes"));
+        }
+    }
+}
+
+bool MasterNodesWidget::startAll(QString& failedMN) {
+    for (CMasternodeConfig::CMasternodeEntry mne : masternodeConfig.getEntries()) {
+        std::string strError;
+        if (!startMN(mne, strError)) {
+            failedMN = QString::fromStdString(mne.getAlias());
+            return false;
```

Comment:
instead of outright returning `false` here, I think we should add `failedMN` to a list and keep going, trying to start subsequent masternode entries.

Reply:
agreed, that would be more consistent with what the RPC command does

Reply:
I did it in that way in the first place but.. have to implement an special type of notification to present on the screen properly the names and causes of failed Masternodes (which is good for a future implementation but not now that we want to release 4.0.1 as soon as possible).

So.. I simplified it. But well, after talk with zebra, what came up until can do the proper notification, with time, is say "N amount of MN failed, M succeed", the rows will be updated and the ones who failed will remain in "missing".